src/harness/testcases/WINNF_FT_S_FDB_testcase.py

```diff
@@ -0,0 +1,1240 @@
+    for relq_resp in relq_responses:
+      self.assertEqual(relq_resp['response']['responseCode'], 0)
+
+    del relq_requests, relq_responses
+    del grant_request_g1, grant_response_g1
+
+    # TOD0
+    # Step 7: Modify the GWBL database to include a modified version of the GWBL 'W'.
+    # 'W' is moved further than 150 kms from FSS site.
+
+    # Step 8: Trigger daily activities
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Step 9: Send Grant Request 'G2'
+    # Forming of 'G2' grants for all the configured grants
+    addCbsdIdsToRequests(cbsd_ids,config['grantRequests'][1])
+    grant_request_g2 = {'grantRequest': config['grantRequests'][1]}
+
+    grant_response_g2 = self._sas.Grant(grant_request_g2)['grantResponse']
+
+    # Check the grant response code is as expected (SUCCESS or INTERFERENCE).
+    for resp in grant_response_g2:
+      self.assertIn(resp['response']['responseCode'], config['expectedResponseCodes'][1])
+
+    del grant_request_g2, grant_response_g2
+
+  def generate_FDB_7_default_config(self, filename):
+    """Generates the WinnForum configuration for FDB.7"""
+
+    # Load devices info
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+
+    # Update the FCC ID of CBSDs with the FCC ID 'F_ID' from FCC ID database.
+    device_a['fccId'] = "F_ID"
+    device_b['fccId'] = "F_ID"
+
+    # Creating conditionals for Cat B devices
+    self.assertEqual(device_b['cbsdCategory'], 'B')
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['installationParam']
+    del device_b['measCapability']
+
+    conditionals = [conditionals_b]
+
+    # Create the actual config.
+    config = {
+        'registrationRequests': [device_a, device_b],
+        'expectedResponseCodes': [(103,), (0,)],
+        'conditionalRegistrationData': conditionals
+        # TODO
+        # Need to add data base configurations
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_FDB_7_default_config)
+  def test_WINNF_FT_S_FDB_7(self, config_filename):
+    """FCC ID Database Update."""
+
+    # Load the configuration file
+    config = loadConfig(config_filename)
+
+    registration_requests = {
+        'registrationRequest': config['registrationRequests']
+    }
+
+    # Step 1: Pre-load conditional registration data for CBSD.
+    if ('conditionalRegistrationData' in config)\
+      and (config['conditionalRegistrationData']):
+      self._sas_admin.PreloadRegistrationData(
```

Comment:
To go with my earlier comment: we can't pre-load here because the SAS UUT doesn't know yet that the FCC ID is valid.

Reply:
FDB.7     Removed conditional registration data from config data structure and   pre-loading .